### DISCUSSION TITLE
Support for React 17 and sendgrid theme

### DISCUSSION THREAD

**Discussion by gautammadaanSG** [twilio-labs/paste#1435] [Q&A] [answered]
Hey Everyone,
We are planning to use React 17 for one of our projects, and I was wondering if Paste is compatible with React 17? OR will it download its own required React version(and not use the cached React 17)?

Secondly, we want to use the Sendgrid theme in our app and I wanted to confirm that it is fully supported and that we can use Paste in Sendgrid UIs with confidence.

**Comment by TheSisb** [comment, marked as answer]
Hi @gautammadaanSG,

Paste requires version `^16.4.0` in your project, but it should be compatible with React 17. Let us know if you encounter any issues and we will address them. 

It won't download two versions of React into your app because Paste marks `react` as a `peerDependency`.  This means it will use whichever version you have installed in your project while letting you know the expected version is `16.14.0`.

You can use the Sendgrid Paste theme with confidence; it is fully supported and encouraged for Sendgrid UIs.
